nixos/modules/services/networking/syncthing.nix

```diff
@@ -484,6 +484,24 @@ in {
               -gui-address=${cfg.guiAddress} \
               -home=${cfg.configDir}
           '';
+          MemoryDenyWriteExecute = true;
+          NoNewPrivileges = true;
+          PrivateDevices = true;
+          PrivateMounts = true;
+          PrivateTmp = true;
+          PrivateUsers = true;
+          ProtectControlGroups = true;
+          ProtectHostname = true;
```

Comment:
More a curious question: Can the process change its hostname in first place, if it isn't run as root?

Reply:
You can do it with some capabilities, at least with `CAP_SYS_ADMIN`. That said, this capability is blocked by our current `CapabilityBoundingSet`.

Better being safe than sorry.

Reply:
Makes sense, thanks!